src/octree/mod.rs

```diff
@@ -302,50 +300,35 @@ impl Octree {
         })
     }
 
-    /// Returns the ids of all nodes that cut or are fully contained in 'aabb'.
-    pub fn points_in_box<'a>(&'a self, bbox: &'a Aabb3<f64>) -> FilteredPointsIterator<'a> {
-        let intersects = |aabb: &Aabb3<f64>| bbox.intersects(aabb);
-        let node_ids = intersecting_node_ids(self, &intersects);
-        let filter_func = Box::new(move |p: &Point| bbox.contains(&Point3::from_vec(p.position)));
-        FilteredPointsIterator::new(&self, node_ids, filter_func)
-    }
-
-    pub fn points_in_obb<'a>(&'a self, obb: &'a Obb<f64>) -> FilteredPointsIterator<'a> {
-        let intersects = |aabb: &Aabb3<f64>| obb.intersects(aabb);
-        let node_ids = intersecting_node_ids(self, &intersects);
-        let filter_func = Box::new(move |p: &Point| obb.contains(&Point3::from_vec(p.position)));
-        FilteredPointsIterator::new(&self, node_ids, filter_func)
-    }
-
-    pub fn points_in_oriented_beam<'a>(
+    pub fn nodes_in_location<'a>(
         &'a self,
-        beam: &'a OrientedBeam,
-    ) -> FilteredPointsIterator<'a> {
-        let intersects = |aabb: &Aabb3<f64>| beam.intersects(aabb);
-        let node_ids = intersecting_node_ids(self, &intersects);
-        let filter_func = Box::new(move |p: &Point| beam.contains(&Point3::from_vec(p.position)));
-        FilteredPointsIterator::new(&self, node_ids, filter_func)
+        location: &PointLocation,
+    ) -> NodeIdsIterator<'a, impl Fn(&NodeId, &'a Octree) -> bool> {
+        let container = location.get_point_culling();
+        let filter_func = move |node_id: &NodeId, octree: &Octree| -> bool {
+            let current = &octree.nodes[&node_id];
+            container.intersects(&current.bounding_cube.to_aabb3())
+        };
+        NodeIdsIterator::new(&self, filter_func)
     }
 
-    pub fn points_in_frustum<'a>(
+    /// Returns the ids of all nodes that cut or are fully contained in 'aabb'.
```

Comment:
Nit: Let's be consistent with two vs three slashes

Reply:
this was meant to be a minimal doc comment for the method points_in_node, do we use doc comment at all?